### DISCUSSION TITLE
routing status 2 despite solving the problem

### DISCUSSION THREAD

**Discussion by prasanik** [google/or-tools#3827] [Routing (and legacy CP) questions] [answered]
I think I've found a possible bug. Solving a VRP problem using...
```py
solution = routing.SolveWithParameters(search_parameters)
if solution:
    print_solution(....)
else:
    print('No solution found !')
print("Solver status: ", routing.status())
```
It solves the problem, goes into print_solution, prints out the route and still reports the status as 2. Please note that it doesn't go into else part and doesn't print 'No solution found !'.
Does this mean if solution, then I can simply assume that a solution is found and don't need to check the solver status at all? Or is it a bug?
If it helps, when I try with around 100 nodes,  I see this behaviour. But not with 62. Haven't tested with lot of other numbers yet.

**Comment by Regista6** [comment, marked as answer]
Status 2 is actually `ROUTING_PARTIAL_SUCCESS_LOCAL_OPTIMUM_NOT_REACHED`.
https://github.com/google/or-tools/blob/5425dedcfbb22cb74c636c1374a9b5ad684b1eb5/ortools/constraint_solver/routing.h#L246-L266

**Reply by prasanik** [reply]
I was going by the docs here.. https://developers.google.com/optimization/routing/routing_options
Do you mean the documentation is not updated with this? And 2 is the value I should expect when I'm setting a time limit for a large problem?

**Reply by Mizux** [reply]
I think the doc is out of date thx for the report -> I will create an issue to track it (#3828)

**Reply by prasanik** [reply]
cool... thanks.
